packages/open-next/src/build.ts

```diff
@@ -507,7 +508,12 @@ function removeCachedPages(outputPath: string, packagePath: string) {
           file.endsWith(".json") ||
           file.endsWith(".rsc") ||
           file.endsWith(".meta") ||
-          (file.endsWith(".html") && !htmlPages.has(file))
+          (file.endsWith(".html") &&
+            // do not remove static HTML files
+            !htmlPages.has(file) &&
+            // do not remove HTML files with "[param].html" format
+            // b/c they are used for "fallback:true" pages
+            !isFallbackTruePage.test(file))
```

Comment:
@conico974 I reworked these few lines. They are less performant than the initial version (okay since this is run at built time), but hopefully a bit more readable.

Will merge once u give green flag.